textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/Span.java

```diff
@@ -3,22 +3,40 @@
 public class Span {
     //The name of the field (in the tuple) where this span is present
     private String fieldName;
-    //The start of the span
+    //The start of the span. It is the position of the first character of span in the document.
     private int start;
-    //The end of the span
+    //The end of the span.It is the position of the first character of span in the document
     private int end;
     //The key we are searching for eg: regex
     private String key;
     //The value matching the key
     private String value;
-    
-    
-    public Span(String fieldName, int start, int end, String key, String value) {
+    // The token position of the span
+    private int tokenOffset;
+
+    /*
+    Example:
+        Value = "The quick brown fox jumps over the lazy dog"
+        Now the Span for brown should be
+        start = 10 : position of character 'b'
+        end = 15 : position of character 'n'

```

Comment:
Hi guys. @sandeepreddy602 @akshaybetala  I'm testing Lucene's position and offset, but I have some problems here. Have you successfully got positions from Lucene? Can you please take a look at the code and see what's wrong?
Here's code, I'm following the [official documentation](https://lucene.apache.org/core/5_5_0/core/org/apache/lucene/index/package-summary.html#positions)
```java
FieldType ft = new FieldType(TextField.TYPE_STORED);
ft.setStoreTermVectors(true);
ft.setStoreTermVectorPositions(true);
ft.setStoreTermVectorOffsets(true);
ft.setStoreTermVectorPayloads(true);

Document doc = new Document();
doc.add(new Field("title", "some title here title1", ft));
doc.add(new Field("content", "keyword data content number", ft));
writer.addDocument(doc);

// ......

SpanTermQuery spanQuery = new SpanTermQuery(new Term("content", "keyword"));

IndexReader reader = DirectoryReader.open(index);
IndexSearcher searcher = new IndexSearcher(reader);

TopDocs docs = searcher.search(spanQuery, reader.numDocs());
ScoreDoc[] hits = docs.scoreDocs;

for (int i = 0; i < hits.length; ++i) {
	int docID = hits[i].doc;
	System.out.println("docID: " + docID);
	System.out.println("content: " + searcher.doc(docID).getField("content").stringValue());
	
	Terms terms = reader.getTermVector(hits[i].doc, "content");
	TermsEnum termsEnum = terms.iterator();
	PostingsEnum postings = termsEnum.postings(null, PostingsEnum.PAYLOADS | PostingsEnum.OFFSETS);
	while ((docID = postings.nextDoc()) != DocIdSetIterator.NO_MORE_DOCS) {
		System.out.println(docID);

		int freq = postings.freq();
	   for (int j = 0; j < freq; j++) {
	      System.out.println(postings.nextPosition());
	      System.out.println(postings.startOffset());
	      System.out.println(postings.endOffset());
	      System.out.println(postings.getPayload());
	   }
	}
}
```

And in line
```java
PostingsEnum postings = termsEnum.postings(null, PostingsEnum.PAYLOADS | PostingsEnum.OFFSETS);
```
I got
```
java.lang.ArrayIndexOutOfBoundsException: -1 at 
org.apache.lucene.codecs.compressing.CompressingTermVectorsReader$TVTermsEnum.postings(CompressingTermVectorsReader.java:949)
```